frontend/src/pages/RunDetails.test.tsx

```diff
@@ -668,6 +668,50 @@ describe('RunDetails', () => {
     expect(tree.state('selectedNodeDetails')).toHaveProperty('phaseMessage', undefined);
   });
 
+  it('displays a spinner if graph is not defined and run has not finished', async () => {
+    const unfinishedRun = {
+      pipeline_runtime: {
+        // No graph
+        workflow_manifest: '{}',
+      },
+      run: {
+        id: 'test-run-id',
+        name: 'test run',
+        // Run has not finished
+        status: 'Running',
```

Comment:
Can we repeat this test for other unfinished statuses?

Reply:
Done